test/common/reference_data_wrangler.hpp

```diff
@@ -92,10 +93,13 @@ auto gen_fourier_data(portfft::descriptor<Scalar, Domain>& desc, portfft::detail
       "  if (is_complex):\n"
       "    inData = inData + 1j * rng.uniform(-1, 1, dataGenDims).astype(scalar_type)\n"
       "  outData = np.fft.fftn(inData, axes=range(1, len(dims) + 1))\n"
-      "  inData.reshape(-1, 1)\n"
-      "  outData.reshape(-1, 1)\n"
-      "  inData = inData.astype(complex_type)\n"
       "  outData = outData.astype(complex_type)\n"
+      "  if (not is_complex):\n"
+      "    fft_len = np.prod(dims)\n"
+      "    bwd_real_len = (fft_len // dims[-1]) * (dims[-1] // 2  + 1)\n"
+      "    # For each batch halve the size of the last FFT dimension\n"
+      "    outData = outData.reshape([batch, fft_len])[:, :bwd_real_len]\n"
```

Comment:
Alternatively we could call np.rfftn to directly calculate real fft. I think that would result in simpler code.

Reply:
Good point, done in https://github.com/codeplaysoftware/portFFT/pull/131/commits/cbb517b9d27b2fdc4159b7f9467ba1ddbbe2a47b

Reply:
Resolving since you have approved